.github/workflows/fips-checksums.yml

```diff
@@ -102,18 +112,80 @@ jobs:
       - uses: actions/checkout@v4
         with:
           path: source
+          persist-credentials: false
       - name: config
         run: ../source/config --banner=Configured $BUILD_OPTS && perl configdata.pm --dump
         working-directory: ./build
       - name: make
         run: make -s -j4
         working-directory: ./build
-      - name: abidiff
-        run: abidiff --headers-dir1 build-pristine/include/openssl --headers-dir2 build/include/openssl --drop-private-types ./build-pristine/libcrypto.so ./build/libcrypto.so && abidiff --headers-dir1 build-pristine/include/openssl --headers-dir2 build/include/openssl --drop-private-types ./build-pristine/libssl.so ./build/libssl.so && touch ./artifact/abi_unchanged || ( touch ./artifact/abi_changed ; echo ABI CHANGED )
-      - name: save PR number
-        run: echo ${{ github.event.number }} > ./artifact/pr_num
-      - name: save artifact
-        uses: actions/upload-artifact@v4
+      - id: abi_result
+        name: abidiff (record status)
+        run: |
+          set +e
+          abidiff --headers-dir1 build-pristine/include/openssl --headers-dir2 build/include/openssl --drop-private-types ./build-pristine/libcrypto.so ./build/libcrypto.so \
+          && abidiff --headers-dir1 build-pristine/include/openssl --headers-dir2 build/include/openssl --drop-private-types ./build-pristine/libssl.so ./build/libssl.so
+          rc=$?
+          if [ $rc -eq 0 ]; then
+            echo "abi_status=unchanged" >> "$GITHUB_OUTPUT"
+          else
+            echo "abi_status=changed" >> "$GITHUB_OUTPUT"
+            echo "ABI CHANGED"
+          fi
+          exit 0
+
+  apply-label:
+    permissions:
+      contents: read
+      pull-requests: write
+    needs: [compute-checksums, compute-abidiff]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Apply/Remove labels (github-script)
+        uses: actions/github-script@v7
+        env:
+          PR_NUM: ${{ github.event.number }}
+          FIPS_STATUS: ${{ needs.compute-checksums.outputs.fips_status }}
```

Comment:
Can you please explain to me how moving the labelling from workflow trigger to on-PR workflow can work? Will the GH token in the on-PR workflow have sufficient permissions to actually apply the label changes?? And if so, how does this prevent anybody from modifying the workflow in their PR to mess with the labelling???

I assume I am missing something here.

Reply:
Here is the test run of this workflow

https://github.com/quarckster/openssl-fork/pull/46

https://github.com/quarckster/openssl-fork/actions/runs/18743812280